cmd/clusterctl/pkg/client/repository/template.go

```diff
@@ -122,11 +98,17 @@ func newTemplate(options newTemplateOptions) (*template, error) {
 	objs = fixTargetNamespace(objs, options.targetNamespace)
 
 	return &template{
-		Provider:        options.provider,
-		version:         options.version,
-		flavor:          options.flavor,
 		variables:       variables,
 		targetNamespace: options.targetNamespace,
 		objs:            objs,
 	}, nil
 }
+
+// NewTemplate returns a new template object
+func NewTemplate(yaml []byte, configVariablesClient config.VariablesClient, targetNamespace string) (*template, error) {
```

Comment:
Do we need both a private and a public constructor?

Reply:
Removed!